spec/src/main/resources/META-INF/smithy/bsp/bsp.smithy

```diff
@@ -0,0 +1,1380 @@
+
+}
+
+structure TestProvider with [LanguageProvider] {
+
+}
+
+structure InitializedBuildParams {
+
+}
+
+@mixin
+structure MessageParams {
+    /// the message type.
+    @required
+    type: MessageType
+    /// The task id if any.
+    task: TaskId
+    /// The request id that originated this notification.
+    /// The originId field helps clients know which request originated a notification in case several requests are handled by the
+    /// client at the same time. It will only be populated if the client defined it in the request that triggered this notification.
+    originId: RequestId
+    /// The actual message.
+    @required
+    message: String
+}
+
+structure OnBuildShowMessageParams with [MessageParams] {
+
+}
+
+structure LogMessageParams with [MessageParams] {
+
+}
+
+@enumKind("open")
+intEnum MessageType {
+    /// An error message.
+    ERROR = 1
+    /// A warning message.
+    WARNING = 2
+    /// An information message.
+    INFO = 3
+    /// A log message.
+    LOG = 4
+}
+
+structure PublishDiagnosticsParams {
+    /// The document where the diagnostics are published.
+    @required
+    textDocument: TextDocumentIdentifier
+    /// The build target where the diagnostics origin.
+    /// It is valid for one text document to belong to multiple
+    /// build targets, for example sources that are compiled against multiple
+    /// platforms (JVM, JavaScript).
+    @required
+    buildTarget: BuildTargetIdentifier
+    /// The request id that originated this notification.
+    originId: RequestId
+    /// The diagnostics to be published by the client.
+    @required
+    diagnostics: Diagnostics
+    /// Whether the client should clear the previous diagnostics
+    /// mapped to the same `textDocument` and `buildTarget`.
+    @required
+    reset: Boolean
+}
+
+structure Diagnostic {
+    /// The range at which the message applies.
+    @required
+    range: Range
+    /// The diagnostic's severity. Can be omitted. If omitted it is up to the
+    /// client to interpret diagnostics as error, warning, info or hint.
+    severity: DiagnosticSeverity
+    /// The diagnostic's code, which might appear in the user interface.
+    code: Code
+    /// An optional property to describe the error code.
+    codeDescription: CodeDescription
+    /// A human-readable string describing the source of this
+    /// diagnostic, e.g. 'typescript' or 'super lint'.
+    source: String
+    /// The diagnostic's message.
+    @required
+    message: String
+    // Additional metadata about the diagnostic.
+    tags: DiagnosticTags
+    /// An array of related diagnostic information, e.g. when symbol-names within
+    /// a scope collide all definitions can be marked via this property.
+    relatedInformation: DiagnosticRelatedInformationList
+    /// A data entry field that is preserved between a `textDocument/publishDiagnostics` notification
+    // and a `textDocument/codeAction` request.
+    data: Json
+}
+
+structure Position {
```

Comment:
Maybe we do somewhere else, but it might be a good idea to mention if these are 0 or 1 based.

Reply:
Addressed.